app/domain/PrepareUploadService.scala

```diff
@@ -14,12 +14,12 @@ class PrepareUploadService @Inject()(
   configuration: ServiceConfiguration,
   metrics: Metrics) {
 
-  def prepareUpload(settings: UploadSettings): PreparedUpload = {
+  def prepareUpload(settings: UploadSettings, consumingService: Option[String]): PreparedUpload = {
```

Comment:
So as discussed, I think we shouldn't be passing through a constructer parameter that could get an upload into an undesired state.
Possible solutions could be:
1. If `User-Agent` isn't found, pass through from the controller "No-Consuming-Service" or similar, to allow us to debug down the line what has happened.
2. Reject within the controller using case matching or similar (but this risks doubling up the check)
3. Have the filter pass through the validated `User-Agent` as a string, worked example to follow.